using_restricted_array.rb

```diff
@@ -8,28 +8,61 @@
 
 ## Calculates the length of the restricted integer array_size
 def length(array)
-  puts "NOT IMPLEMENTED"
+  length = 0
+  while array[length] != nil
+    length += 1
+  end
+  return length
 end
 
 # Prints each integer values in the array
 def print_array(array)
-  puts "NOT IMPLEMENTED"
+  (0..array.array_size).each do |i|

```

Comment:
Although, array_size was intentionally kept private, you do have access to the length method that you just implemented. :-)